test/e2e/set/scyllacluster/scyllacluster_auth.go

```diff
@@ -0,0 +1,163 @@
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		// TODO: restart should be triggered by the Operator
+		framework.By("Initiating a rolling restart")
+
+		gen, err := restartRack(ctx, f.KubeClient().AppsV1(), sc.Spec.Datacenter.Racks[0], sc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		framework.By("Waiting for the ScyllaCluster to pick up token change")
+
+		rackGenerationUpgraded := func(sc *scyllav1.ScyllaCluster) (bool, error) {
+			stss, err := getStatefulSetsForScyllaCluster(ctx, f.KubeClient().AppsV1(), sc)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			return stss[sc.Spec.Datacenter.Racks[0].Name].Status.ObservedGeneration == gen, nil
+		}
+
+		waitCtx2, waitCtx2Cancel := contextForRollout(ctx, sc)
+		defer waitCtx2Cancel()
+		sc, err = waitForScyllaClusterState(waitCtx2, f.ScyllaClient().ScyllaV1(), sc.Namespace, sc.Name, scyllaClusterRolledOut, rackGenerationUpgraded)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		framework.By("Accepting requests authorized using token from user agent config")
+		checkAgentAuth(ctx, hosts, agentConfig.AuthToken, func(err error) {
+			o.Expect(err).NotTo(o.HaveOccurred())
+		})
+
+		framework.By("Changing auth token in agent config")
+
+		agentConfig.AuthToken = "666"
+		agentConfigData, err = yaml.Marshal(agentConfig)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		agentConfigSecret.Data[naming.ScyllaAgentConfigFileName] = agentConfigData
+		_, err = f.KubeClient().CoreV1().Secrets(f.Namespace()).Update(ctx, agentConfigSecret, metav1.UpdateOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		framework.By("Initiating a rolling restart")
+
+		gen, err = restartRack(ctx, f.KubeClient().AppsV1(), sc.Spec.Datacenter.Racks[0], sc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		framework.By("Waiting for the ScyllaCluster to pick up token change")
+
+		rackGenerationUpgraded = func(sc *scyllav1.ScyllaCluster) (bool, error) {
+			stss, err := getStatefulSetsForScyllaCluster(ctx, f.KubeClient().AppsV1(), sc)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			return stss[sc.Spec.Datacenter.Racks[0].Name].Status.ObservedGeneration == gen, nil
+		}
+
+		waitCtx3, waitCtx3Cancel := contextForRollout(ctx, sc)
+		defer waitCtx3Cancel()
+		sc, err = waitForScyllaClusterState(waitCtx3, f.ScyllaClient().ScyllaV1(), sc.Namespace, sc.Name, scyllaClusterRolledOut, rackGenerationUpgraded)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		framework.By("Accepting requests authorized using token from user agent config")
+		checkAgentAuth(ctx, hosts, agentConfig.AuthToken, func(err error) {
+			o.Expect(err).NotTo(o.HaveOccurred())
+		})
+	})
+})
+
+func checkAgentAuth(ctx context.Context, hosts []string, authToken string, f func(err error)) {
+	cfg := scyllaclient.DefaultConfig(authToken, hosts...)
+
+	client, err := scyllaclient.NewClient(cfg, log.NopLogger)
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	_, err = client.Status(ctx, hosts[0])
+	f(err)
```

Comment:
would it be cleaner to just return the error and handle it on the call site?
`func getScyllaClientStatus(ctx context.Context, hosts []string, authToken string) (Status, error)`